drivers/storage/portworx/component/portworx_basic.go

```diff
@@ -557,23 +557,28 @@ func (c *portworxBasic) createAndMaintainPxSaTokenSecret(cluster *corev1.Storage
 			return err
 		}
 	}
-	needRefresh, err := isTokenRefreshRequired(secret)
+	caCrtUpdated, err := updateCaCrtIfNeeded(secret)
 	if err != nil {
 		return err
 	}
-	if needRefresh {
-		if err := c.refreshTokenSecret(secret, cluster, ownerRef); err != nil {
+	needRefreshToken, err := isTokenRefreshRequired(secret)
```

Comment:
nit: add a updateTokenIfNeeded() wrapper to be consistent with updateCaCrtIfNeeded?